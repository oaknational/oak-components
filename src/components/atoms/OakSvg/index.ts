export { OakSvg } from "./OakSvg";
export type { OakSvgProps, OakSvgNames } from "./OakSvg";
