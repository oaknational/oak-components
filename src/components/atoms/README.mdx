# Atoms

## Rules for components in this folder

- Can import other atoms
- Imports from molecules and organisms are forbidden!
- Generic
- Unstyled
- Avoid embedded logic and state
- No sub-components
  (eg. Box)
