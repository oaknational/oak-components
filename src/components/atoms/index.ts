export * from "./OakBox";
export * from "./OakFlex";
export * from "./OakSpan";
export * from "./OakForm";
export * from "./OakImage";
export * from "./OakTypography";
export * from "./OakHeading";
export * from "./OakP";
export * from "./OakUL";
export * from "./OakOL";
export * from "./OakLI";
export * from "./OakLabel";
export * from "./OakIcon";
export * from "./OakGrid";
export * from "./OakGridArea";
export * from "./OakThemeProvider";
export * from "./OakMaxWidth";
export * from "./OakCloudinaryImage";
export * from "./OakKbd";
export * from "./OakGlobalStyle";
export * from "./OakAnchorTarget";
export * from "./OakFieldSet";
