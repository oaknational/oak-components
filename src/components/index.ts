export * from "./base";
// export * from "./ui"; // uncomment this line once there are components in this folder
// export * from "./integrated"; // uncomment this line once there are components in this folder
