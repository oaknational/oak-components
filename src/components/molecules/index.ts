export * from "./OakFieldError";
export * from "./OakLoadingSpinner";
export * from "./OakPrimaryButton";
export * from "./OakPrimaryInvertedButton";
export * from "./OakSecondaryButton";
export * from "./OakTertiaryButton";
export * from "./OakRadioButton";
export * from "./OakRadioGroup";
export * from "./OakCheckBox";
export * from "./OakTextInput";
export * from "./OakRoundIcon";
export * from "./OakTooltip";
export * from "./OakBackLink";
export * from "./OakBulletList";
export * from "./OakHandDrawnCard";
export * from "./OakHandDrawnCardWithIcon";
export * from "./OakHandDrawnHR";
export * from "./OakPromoTag";
export * from "./OakLessonInfoCard";
export * from "./OakCollapsibleContent";
export * from "./OakLink";
export * from "./OakSecondaryLink";
export * from "./OakDragAndDropInstructions";
export * from "./OakDraggable";
export * from "./OakDroppable";
export * from "./OakDraggableFeedback";
export * from "./OakOutlineAccordion";
export * from "./OakAccordion";
export * from "./OakModal";
export * from "./OakModalCenter";
export * from "./OakCardWithHandDrawnBorder";
export * from "./OakSecondaryButtonAsRadio";
export * from "./OakButtonAsRadioGroup";
