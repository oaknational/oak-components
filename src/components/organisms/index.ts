export * from "./pupil";
export * from "./teacher";
export * from "./OakSubjectIcon";
export * from "./OakPrimaryNavItem";
export * from "./OakPrimaryNav";
export * from "./OakCookieBanner";
export * from "./OakCookieSettingsModal";
export * from "./OakCookieConsentProvider";
export * from "./OakCookieConsent";
