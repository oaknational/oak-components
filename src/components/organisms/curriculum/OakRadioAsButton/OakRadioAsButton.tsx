import React, { useContext, useId, useRef } from "react";
import styled from "styled-components";

import { OakFlex, OakIcon, OakIconName } from "@/components/atoms";
import { InternalCheckBoxLabelHoverDecor } from "@/components/atoms/InternalCheckBoxLabel";
import {
  BaseRadioProps,
  InternalRadio,
} from "@/components/atoms/InternalRadio/InternalRadio";
import { parseColor } from "@/styles/helpers/parseColor";
import { parseBorder } from "@/styles/helpers/parseBorder";
import { parseDropShadow } from "@/styles/helpers/parseDropShadow";
import { parseColorFilter } from "@/styles/helpers/parseColorFilter";
import { RadioContext } from "@/components/molecules/OakRadioGroup/OakRadioGroup";

// Converted to styled-component so it can be used in '&:checked:not(:disabled) + ${StyledOakIcon}' to change svg color.
const StyledOakIcon = styled(OakIcon)``;

const StyledInternalRadio = styled(InternalRadio)<{
  keepIconColor?: boolean;
}>`
  position: absolute;
  opacity: 0;
  pointer-events: none;

  &:checked:not(:disabled) + ${StyledOakIcon} {
    filter: ${(props) =>
      props.keepIconColor ? "none" : parseColorFilter("white")};
  }

  &:checked:not(:disabled) {
    border: ${parseBorder("border-solid-l")};
    border-color: ${parseColor("border-primary")};
  }

  &:checked:disabled {
    border: ${parseBorder("border-solid-l")};
    border-color: ${parseColor("text-disabled")};
  }

  &:hover:not(:disabled) {
    background: ${parseColor("bg-primary")};
  }
`;

const StyledFlexBox = styled(OakFlex)`
  &:has(input:not(:disabled)) {
    cursor: pointer;
  }

  &:has(input:disabled) {
    pointer-events: none;
    cursor: none;
  }

  &:hover:has(input:not(:disabled)) ${InternalCheckBoxLabelHoverDecor} {
    text-decoration: underline;
  }

  &:hover:has(input:not(:disabled)) {
    background-color: ${parseColor("bg-neutral")};
  }

  &:focus-within {
    box-shadow: ${parseDropShadow("drop-shadow-centered-lemon")},
      ${parseDropShadow("drop-shadow-centered-grey")};
  }

  &:has(input:checked:not(:disabled)) {
    background-color: ${parseColor("black")};
    color: ${parseColor("white")};
  }
`;

export type OakRadioAsButtonProps = Omit<
  BaseRadioProps,
  "defaultChecked" | "id"
> & {
  innerRef?: React.RefObject<HTMLInputElement>;
  displayValue: string;
  icon?: OakIconName;
  keepIconColor?: boolean;
  disabled?: HTMLInputElement["disabled"];
  checked?: HTMLInputElement["checked"];
  value?: HTMLInputElement["value"];
  "aria-labelledby"?: React.AriaAttributes["aria-labelledby"];
  "aria-label"?: React.AriaAttributes["aria-label"];
  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
};

/**
 * A radio input styled as a button, to be used within `<OakRadioGroup/>` this is
 * the radio inputs version of `<OakSearchFilterCheckBox/>`
 *
 * ## Events
 * The following callbacks are available for tracking focus events:
 *
 * ### onChange
 *  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
 *
 * ### onFocus
 *   onFocus?: (event: React.FocusEvent<HTMLInputElement>) => void;
 *
 * ### onBlur
 *    onBlur?: (event: React.FocusEvent<HTMLInputElement>) => void;
 *
 * ### onHovered
 *  `onHovered?: (id, value, duration: number) => void;`<br>
 *  called after a mouseEnter and mouseLeave event has happened
 *
 */
export const OakRadioAsButton = (props: OakRadioAsButtonProps) => {
  const id = useId();
  const { value, disabled, innerRef, displayValue, icon, onChange, ...rest } =
    props;
  const { name, onValueUpdated, currentValue } = useContext(RadioContext);

  const defaultRef = useRef<HTMLInputElement>(null);
  const inputRef = innerRef ?? defaultRef;

  const handleContainerClick = (
    e:
      | React.MouseEvent<HTMLDivElement>
      | React.MouseEvent<HTMLInputElement>
      | React.MouseEvent<HTMLLabelElement>,
  ) => {
    const el = e.target as HTMLInputElement;

    if (!el.isEqualNode(inputRef.current)) {
      inputRef.current?.click();
    }
  };

  return (
    <OakFlex $minHeight={"all-spacing-8"} $position={"relative"}>
      <StyledFlexBox
        $borderRadius={"border-radius-s"}
        $borderColor={"border-neutral-lighter"}
        $ba="border-solid-s"
        $background={"white"}
        onClick={handleContainerClick}
        $ph={"inner-padding-s"}
        $pv={"inner-padding-ssx"}
        $gap={"space-between-sssx"}
      >
        <StyledInternalRadio
          {...rest}
          id={id}
          value={value}
          disabled={disabled}
          ref={inputRef}
          onChange={(e) => {
            onValueUpdated?.(e);
            onChange?.(e);
          }}
          name={name}
          checked={currentValue === value}
        />
        {icon && <StyledOakIcon alt="" iconName={icon} />}
        <InternalCheckBoxLabelHoverDecor
          pointerEvents="none"
          htmlFor={id}
          $font={"heading-7"}
          disabled={disabled}
        >
          {displayValue}
        </InternalCheckBoxLabelHoverDecor>
      </StyledFlexBox>
    </OakFlex>
  );
};
