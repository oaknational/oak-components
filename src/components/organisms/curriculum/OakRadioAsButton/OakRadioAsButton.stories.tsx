import React from "react";
import { Meta, StoryObj } from "@storybook/react";

import { OakRadioAsButton } from "./OakRadioAsButton";

import { OakRadioGroup } from "@/components/molecules";

const meta: Meta<typeof OakRadioAsButton> = {
  component: OakRadioAsButton,
  tags: ["autodocs"],
  parameters: {
    backgrounds: {
      default: "light",
    },
    controls: {
      include: ["disabled", "checked", "displayValue", "value", "icon"],
    },
  },
};
export default meta;

type Story = StoryObj<typeof OakRadioAsButton>;

export const Default: Story = {
  render: (args) => <OakRadioAsButton {...args} />,
  args: {
    value: "a test value",
    displayValue: "Art and design",
    icon: "subject-art",
  },
  parameters: {
    controls: {
      include: ["disabled", "checked", "displayValue", "value", "icon"],
    },
  },
};

export const NoIcon: Story = {
  render: (args) => <OakRadioAsButton {...args} />,
  args: {
    value: "a test value",
    displayValue: "Art and design",
    "aria-label": "Art and design",
  },
  parameters: {
    controls: {
      include: ["disabled", "checked", "displayValue", "value", "icon"],
    },
  },
};

export const WithAriaLabel: Story = {
  args: {
    name: "radio-1",
    value: "Option 1",
    displayValue: "Art and design",
    icon: "subject-art",
  },
};

export const WithAriaLabelledBy: Story = {
  render: () => (
    <>
      <h2 id="subject-label">Choose a subject</h2>
      <OakRadioGroup name="test" aria-labelledby="subject-label">
        <OakRadioAsButton
          value="Option 1"
          displayValue="Biology"
          icon="subject-biology"
        />
        <OakRadioAsButton
          value="Option 1"
          displayValue="Biology"
          icon="subject-biology"
        />
      </OakRadioGroup>
    </>
  ),
};

export const MultipleOptions: Story = {
  render: () => (
    <OakRadioGroup
      name="subjects"
      aria-label="Choose a subject"
      $flexWrap={"wrap"}
    >
      <OakRadioAsButton
        value="art"
        displayValue="Art and design"
        icon="subject-art"
      />
      <OakRadioAsButton
        value="biology"
        displayValue="Biology"
        icon="subject-biology"
      />
      <OakRadioAsButton
        value="chemistry"
        displayValue="Chemistry"
        icon="subject-chemistry"
      />
      <OakRadioAsButton
        value="physics"
        displayValue="Physics"
        icon="subject-physics"
      />
      <OakRadioAsButton
        value="computing"
        displayValue="Computing"
        icon="subject-computing"
      />
    </OakRadioGroup>
  ),
};

export const KeepIconColor: Story = {
  render: (args) => {
    const { "aria-labelledby": _, ...restArgs } = args;
    return (
      <OakRadioGroup
        name="test"
        aria-label="Choose a subject"
        $flexWrap={"wrap"}
      >
        <OakRadioAsButton
          {...restArgs}
          displayValue="Art and design"
          icon="subject-art"
        />
        <OakRadioAsButton
          {...restArgs}
          displayValue="Biology"
          icon="subject-biology"
        />
        <OakRadioAsButton
          {...restArgs}
          displayValue="Chemistry"
          icon="subject-chemistry"
        />
        <OakRadioAsButton
          {...restArgs}
          displayValue="Physics"
          icon="subject-physics"
        />
        <OakRadioAsButton
          {...restArgs}
          displayValue="Computing"
          icon="subject-computing"
        />
      </OakRadioGroup>
    );
  },
  args: {
    value: "a test value",
    displayValue: "Lessons",
    icon: "teacher-unit",
    keepIconColor: true,
  },
  parameters: {
    controls: {
      include: ["disabled", "checked", "displayValue", "value", "icon"],
    },
  },
};
