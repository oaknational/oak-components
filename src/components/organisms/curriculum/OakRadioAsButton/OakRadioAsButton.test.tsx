import "@testing-library/jest-dom";
import React from "react";
import { create } from "react-test-renderer";
import { fireEvent } from "@testing-library/react";

import { OakRadioAsButton } from "./OakRadioAsButton";

import renderWithTheme from "@/test-helpers/renderWithTheme";
import { OakThemeProvider } from "@/components/atoms";
import { oakDefaultTheme } from "@/styles";
import { OakRadioGroup } from "@/components/molecules";

describe("OakRadioAsButton", () => {
  it("renders a radio", () => {
    const { getByRole } = renderWithTheme(
      <OakRadioAsButton
        name="radio-1"
        value="history"
        icon={"subject-history"}
        displayValue="History"
      />,
    );
    expect(getByRole("radio")).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const tree = create(
      <OakThemeProvider theme={oakDefaultTheme}>
        <OakRadioAsButton
          name="radio-1"
          value="history"
          icon={"subject-history"}
          displayValue="History"
        />
        ,
      </OakThemeProvider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("has a label", () => {
    const { getByLabelText } = renderWithTheme(
      <OakRadioAsButton
        name="radio-1"
        displayValue="Option 1"
        value="1"
        icon={"subject-history"}
      />,
    );
    expect(getByLabelText("Option 1")).toBeInTheDocument();
  });

  it("renders without an icon when only label provided", () => {
    const { container } = renderWithTheme(
      <OakRadioAsButton
        name="radio-1"
        value="Option 1"
        displayValue="Option 1"
      />,
    );

    const iconElement = container.querySelector("img");
    expect(iconElement).not.toBeInTheDocument();
  });

  it("renders the correct icon when a valid icon name is provided", async () => {
    const { container } = renderWithTheme(
      <OakRadioAsButton
        name="radio-1"
        value="Option 1"
        displayValue="Option 1"
        icon="subject-history"
      />,
    );

    const iconElement = container.querySelector("img");
    expect(iconElement).toBeInTheDocument();
  });

  it("renders both icon and label", async () => {
    const { container, getByText } = renderWithTheme(
      <OakRadioAsButton
        name="radio-1"
        value="Option 1"
        displayValue="Display Text"
        icon="subject-history"
      />,
    );

    const iconElement = container.querySelector("img");
    expect(iconElement).toBeInTheDocument();

    const labelElement = getByText("Display Text");
    expect(labelElement).toBeInTheDocument();
  });

  it("calls onChange when clicked", () => {
    const onChange = jest.fn();
    const { getAllByRole } = renderWithTheme(
      <OakRadioGroup name="test">
        <OakRadioAsButton
          value="Option 1"
          onChange={onChange}
          icon={"subject-history"}
          displayValue="Option 1"
        />
        <OakRadioAsButton
          value="Option 2"
          onChange={onChange}
          icon={"subject-biology"}
          displayValue="Option 2"
        />
      </OakRadioGroup>,
    );
    const radios = getAllByRole("radio");
    expect(radios).toHaveLength(2);

    // Click each twice to make sure we don't register clicks twice
    radios[0]!.click();
    radios[0]!.click();
    radios[1]!.click();
    radios[1]!.click();
    expect(onChange).toHaveBeenCalledTimes(2);
  });

  it("should update checked attribute in DOM", () => {
    const { getAllByRole } = renderWithTheme(
      <OakRadioGroup name="test-group" value="option1">
        <OakRadioAsButton value="option1" displayValue="Option 1" />
        <OakRadioAsButton value="option2" displayValue="Option 2" />
      </OakRadioGroup>,
    );

    const radios = getAllByRole("radio") as HTMLInputElement[];
    fireEvent.click(radios[0]!);

    expect(radios[0]!.checked).toBe(true);
  });

  it("should uncheck previous selection", () => {
    const { getAllByRole } = renderWithTheme(
      <OakRadioGroup name="test-group">
        <OakRadioAsButton value="option1" displayValue="Option 1" />
        <OakRadioAsButton value="option2" displayValue="Option 2" />
      </OakRadioGroup>,
    );

    const radios = getAllByRole("radio") as HTMLInputElement[];

    fireEvent.click(radios[0]!);
    fireEvent.click(radios[1]!);
    expect(radios[0]).not.toBeChecked();
    expect(radios[1]).toBeChecked();
  });

  it("should maintain single value in group", () => {
    let groupValue = "";
    const { getAllByRole } = renderWithTheme(
      <OakRadioGroup
        name="test-group"
        onChange={(e) => (groupValue = e.target.value)}
      >
        <OakRadioAsButton value="option1" displayValue="Option 1" />
        <OakRadioAsButton value="option2" displayValue="Option 2" />
      </OakRadioGroup>,
    );

    fireEvent.click(getAllByRole("radio")[0]!);
    fireEvent.click(getAllByRole("radio")[1]!);

    expect(groupValue).toBe("option2");
  });

  it("should pass the correct value in onChange", () => {
    let lastValue = "";
    const { getAllByRole } = renderWithTheme(
      <OakRadioGroup
        name="test-group"
        onChange={(e) => (lastValue = e.target.value)}
      >
        <OakRadioAsButton value="option1" displayValue="Option 1" />
        <OakRadioAsButton value="option2" displayValue="Option 2" />
      </OakRadioGroup>,
    );

    fireEvent.click(getAllByRole("radio")[1]!);
    expect(lastValue).toBe("option2");
  });

  it("inputs to all have name of radio group", () => {
    const onChange = jest.fn();
    const { getAllByRole, getByRole } = renderWithTheme(
      <OakRadioGroup name="test">
        <OakRadioAsButton
          value="Option 1"
          onChange={onChange}
          icon={"subject-history"}
          displayValue="Option 1"
        />
        <OakRadioAsButton
          value="Option 2"
          onChange={onChange}
          icon={"subject-biology"}
          displayValue="Option 2"
        />
      </OakRadioGroup>,
    );
    const radioGroup = getByRole("radiogroup");
    const radios = getAllByRole("radio");
    expect(radios).toHaveLength(2);
    expect(radioGroup);
    expect(radios[0]).toHaveAttribute("name", "test");
    expect(radios[1]).toHaveAttribute("name", "test");
  });

  it("radio group value sets the correct input to checked", () => {
    const { getAllByRole, getByRole } = renderWithTheme(
      <OakRadioGroup name="test" value="option_2">
        <OakRadioAsButton
          value="option_1"
          icon={"subject-history"}
          displayValue="Option 1"
        />
        <OakRadioAsButton
          value="option_2"
          icon={"subject-biology"}
          displayValue="Option 2"
        />
        <OakRadioAsButton
          value="option_3"
          icon={"subject-biology"}
          displayValue="Option 3"
        />
      </OakRadioGroup>,
    );
    const radioGroup = getByRole("radiogroup");
    const radios = getAllByRole("radio");
    expect(radios).toHaveLength(3);
    expect(radioGroup);
    expect(radios[0]).not.toBeChecked();
    expect(radios[1]).toBeChecked();
    expect(radios[2]).not.toBeChecked();
  });

  it("calls onFocus when focused", () => {
    const onFocus = jest.fn();
    const { getByRole } = renderWithTheme(
      <OakRadioAsButton
        name="radio-1"
        value="Option 1"
        onFocus={onFocus}
        icon={"subject-history"}
        displayValue="Option 1"
      />,
    );
    getByRole("radio").focus();
    expect(onFocus).toHaveBeenCalled();
  });

  it("calls onBlur when blurred", () => {
    const onBlur = jest.fn();
    const { getByRole } = renderWithTheme(
      <OakRadioAsButton
        name="radio-1"
        value="Option 1"
        onBlur={onBlur}
        icon={"subject-history"}
        displayValue="Option 1"
      />,
    );
    getByRole("radio").focus();
    getByRole("radio").blur();
    expect(onBlur).toHaveBeenCalled();
  });

  it("calls onHovered when mouse entered and left", () => {
    const onHovered = jest.fn();
    const { getByRole } = renderWithTheme(
      <OakRadioAsButton
        name="radio-1"
        value="Option 1"
        onHovered={onHovered}
        icon={"subject-history"}
        displayValue="Option 1"
      />,
    );
    fireEvent.mouseEnter(getByRole("radio"));
    fireEvent.mouseLeave(getByRole("radio"));
    expect(onHovered).toHaveBeenCalled();
  });

  it("is not clickable when disabled", () => {
    const onChange = jest.fn();
    const { getByRole } = renderWithTheme(
      <OakRadioAsButton
        name="radio-1"
        value="Option 1"
        onChange={onChange}
        disabled
        icon={"subject-history"}
        displayValue="Option 1"
      />,
    );
    getByRole("radio").click();
    expect(onChange).not.toHaveBeenCalled();
  });

  it("clicking radio triggers onChange", async () => {
    const onChange = jest.fn();
    const { getAllByRole } = renderWithTheme(
      <OakRadioGroup name="test" onChange={onChange}>
        <OakRadioAsButton
          value="option_1"
          icon={"subject-history"}
          displayValue="Option 1"
        />
        <OakRadioAsButton
          value="option_2"
          icon={"subject-biology"}
          displayValue="Option 2"
        />
        <OakRadioAsButton
          value="option_3"
          icon={"subject-biology"}
          displayValue="Option 3"
        />
      </OakRadioGroup>,
    );
    getAllByRole("radio")[1]?.click();
    expect(onChange).toHaveBeenCalledTimes(1);
    expect(onChange).toHaveBeenCalledWith(
      expect.objectContaining({
        target: expect.objectContaining({ value: "option_2" }),
      }),
    );
  });

  it("renders with aria-label attribute", () => {
    const { getByRole } = renderWithTheme(
      <OakRadioAsButton
        name="radio-1"
        value="history"
        displayValue="History"
        icon={"subject-history"}
        aria-label="History"
      />,
    );

    const radioButton = getByRole("radio");
    expect(radioButton).toHaveAttribute("aria-label", "History");
  });
});
