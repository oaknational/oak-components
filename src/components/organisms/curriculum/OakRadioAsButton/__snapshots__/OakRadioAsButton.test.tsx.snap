// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OakRadioAsButton matches snapshot 1`] = `
[
  .c0 {
  position: relative;
  min-height: 2.5rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c2 {
  padding-left: 0.75rem;
  padding-right: 0.75rem;
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
  background: #ffffff;
  border: 0.063rem solid;
  border-color: #cacaca;
  border-radius: 0.25rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c2:hover {
  cursor: pointer;
}

.c7 {
  position: relative;
  width: 2rem;
  min-width: 2rem;
  height: 2rem;
  min-height: 2rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c8 {
  object-fit: contain;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.25rem;
}

.c10 {
  font-family: --var(google-font),Lexend,sans-serif;
  font-weight: 600;
  font-size: 1rem;
  line-height: 1.25rem;
  -webkit-letter-spacing: 0.0115rem;
  -moz-letter-spacing: 0.0115rem;
  -ms-letter-spacing: 0.0115rem;
  letter-spacing: 0.0115rem;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c12 {
  pointer-events: none;
  cursor: pointer;
}

.c5 {
  -webkit-appearance: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  cursor: pointer;
  margin: 0;
  outline: none;
  border: 0.125rem solid;
  border-color: #222222;
  border-radius: 0.125rem;
}

.c5:checked {
  background: #222222;
}

.c5:disabled {
  pointer-events: none;
}

.c6 {
  position: absolute;
  opacity: 0;
  pointer-events: none;
}

.c6:checked:not(:disabled) {
  border: 0.188rem solid;
  border-color: #222222;
}

.c6:checked:disabled {
  border: 0.188rem solid;
  border-color: #808080;
}

.c6:hover:not(:disabled) {
  background: #ffffff;
}

.c4:has(input:not(:disabled)) {
  cursor: pointer;
}

.c4:has(input:disabled) {
  pointer-events: none;
  cursor: none;
}

.c4:hover:has(input:not(:disabled)) .c9 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c4:hover:has(input:not(:disabled)) {
  background-color: #f2f2f2;
}

.c4:focus-within {
  box-shadow: 0 0 0 0.125rem rgba(255,229,85,100%), 0 0 0 0.3rem rgba(87,87,87,100%);
}

.c4:has(input:checked:not(:disabled)) {
  background-color: #222222;
  color: #ffffff;
}

@media (hover:hover) {
  .c12:hover {
    -webkit-text-decoration: underline;
    text-decoration: underline;
  }
}

@media (hover:hover) {
  .c5:hover:not(:disabled) {
    border-color: #222222;
  }
}

<div
    className="c0 c1"
  >
    <div
      className="c2 c3 c4"
      onClick={[Function]}
    >
      <input
        checked={false}
        className="c5 c6"
        id=":r0:"
        name="default"
        onChange={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        type="radio"
        value="history"
      />
      <span
        className="c7 "
      >
        <img
          alt=""
          className="c8"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/v1728568980/subject-icons/history.svg"
          style={
            {
              "bottom": 0,
              "color": "transparent",
              "height": "100%",
              "left": 0,
              "objectFit": undefined,
              "objectPosition": undefined,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": "100%",
            }
          }
        />
      </span>
      <label
        className="c9 c10 c11 c12"
        htmlFor=":r0:"
        pointerEvents="none"
      >
        History
      </label>
    </div>
  </div>,
  ",",
]
`;
